toolkit/tools/pkg/imagecustomizerlib/customizeboot.go

```diff
@@ -170,40 +159,90 @@ func appendKernelCommandLineArguments(inputGrubCfgContent string, extraCommandLi
 	return outputGrubCfgContent, nil
 }
 
-func replaceKernelCommandLineArgumentValue(inputGrubCfgContent string, name string, value string) (outputGrubCfgContent string, oldValue string, err error) {
+// Tries to find the specified kernel CLI arg. Does not fail if the arg is not found.
+//
+// Returns:
+// - startIndex: The string index where the arg was found. Or -1 if not found.
+// - endIndex: The string index of the end of the arg. Or -1 if not found.
+// - value: The value of the arg. Or "" if not found.
+func findKernelCommandLineArgumentArg(grub2Config string, name string) (int, int, string, error) {
```

Comment:
edge case, but what happens if the same key is used twice? Some cmdline args support that additively (crash kernel size for example), while others just take the last in the list.

Reply:
This code actually has a lot of problems, such as being unable to handle escape sequences and single/double quoted strings. So, only really aiming for good enough at the moment.